src/wp-includes/feed.php

```diff
@@ -394,25 +394,30 @@ function get_the_category_rss( $type = null ) {
 
 	if ( ! empty( $categories ) ) {
 		foreach ( (array) $categories as $category ) {
-			$cat_names[] = sanitize_term_field( 'name', $category->name, $category->term_id, 'category', $filter );
+			$cat_name = sanitize_term_field( 'name', $category->name, $category->term_id, 'category', $filter );
+			$cats[ $cat_name ] = $category;
 		}
 	}
 
 	if ( ! empty( $tags ) ) {
 		foreach ( (array) $tags as $tag ) {
-			$cat_names[] = sanitize_term_field( 'name', $tag->name, $tag->term_id, 'post_tag', $filter );
+			$cat_name = sanitize_term_field( 'name', $tag->name, $tag->term_id, 'post_tag', $filter );
+			$cats[ $cat_name ] = $tag;
 		}
 	}
 
-	$cat_names = array_unique( $cat_names );
-
-	foreach ( $cat_names as $cat_name ) {
+	foreach ( $cats as $cat_name => $category ) {
 		if ( 'rdf' === $type ) {
 			$the_list .= "\t\t<dc:subject><![CDATA[$cat_name]]></dc:subject>\n";
 		} elseif ( 'atom' === $type ) {
-			$the_list .= sprintf( '<category scheme="%1$s" term="%2$s" />', esc_attr( get_bloginfo_rss( 'url' ) ), esc_attr( $cat_name ) );
+			$blog_url = apply_filters( 'get_bloginfo_rss', get_bloginfo( 'url' ) );
```

Comment:
Add docblock for the filter

Reply:
Should be handled now, with commit 737a03568d